src/main/ApplicationUtils.cpp

```diff
@@ -30,75 +32,125 @@
 namespace stellar
 {
 
-int
-runWithConfig(Config cfg, std::optional<CatchupConfiguration> cc)
+bool
+canRebuildFromBuckets(uint32_t startAtLedger, uint32_t lcl)
 {
-    VirtualClock::Mode clockMode = VirtualClock::REAL_TIME;
+    // Number of streaming ledgers ahead of LCL. Core will
+    // rebuild the existing state if the difference between the start
+    // ledger and LCL is within this window.
+    uint32_t const RESTORE_STATE_LEDGER_WINDOW = 10;
+    // Do not rebuild genesis state
+    bool isGenesis = lcl == LedgerManager::GENESIS_LEDGER_SEQ;
+    return !isGenesis && startAtLedger >= lcl &&
+           startAtLedger - lcl <= RESTORE_STATE_LEDGER_WINDOW;
+}
+
+void
+setupMinimalDB(Config const& cfg, uint32_t startAtLedger)
```

Comment:
I like the term "minimal DB" you're using here -- I would repeat / reuse / surface this term to the user. It is clear and can be extended into the phrase "minimal DB for in-memory mode".

Reply:
(Maybe also extend the name of this function so any reader immediately knows it's about in-memory mode: `setupMinimalDBForInMemoryMode`